piecedirectory/piecedirectory.go

```diff
@@ -109,6 +113,11 @@ func (ps *PieceDirectory) GetPieceDeals(ctx context.Context, pieceCid cid.Cid) (
 		return nil, fmt.Errorf("listing deals for piece %s: %w", pieceCid, err)
 	}
 
+	//TODO: confirm this change in behaviour is ok
+	if len(deals) == 0 {
```

Comment:
Wouldn't clean up prevent this from hapenning? If a piece has no deals, it should be cleaned up.

Reply:
How can a `piece` have no deals? I don't think this is possible.

I am not aware of any case where `err == nil && len(deals) == 0` but the one where we don't have an unsealed copy.

Reply:
At the moment GetPieceDeals doesn't know anything about sealing - it just knows about the deals associated with a piece.
Note that it will return [ErrNotFound](https://github.com/filecoin-project/boost/blob/052c81a0fbd48e78d018bbaa5bf066202dff6a52/extern/boostd-data/svc/types/types.go#L14) if there is no metadata for a given piece cid (ie there are no deals and no index).

Reply:
At some point in the stack, we should detect that there are no unsealed deals. I am thinking this is a good place, because the API of returning `nil` error and `nil` DealInfo, doesn't make sense and is not expected.

When calling this API I expect to either get deals, or get an error basically.

Reply:
@dirkmc you're right this is the wrong place to add this.